web-scraper/scraping_cart.ipynb

```diff
@@ -0,0 +1,139 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import codecs\n",
+    "import json\n",
+    "import re\n",
+    "import requests\n",
+    "from bs4 import BeautifulSoup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from here on we are going to use our own html code because we can't seem to open and add something to shopping cart through our script\n",
+    "our_page = codecs.open(\"fake_cart2.html\", 'r') \n",
+    "soup = BeautifulSoup(our_page, 'html.parser')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[bs4.element.Tag]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# to check what kind of data we have in our file; seems like we only have tags\n",
+    "[type(item) for item in list(soup.children)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# assign html to our first index to get data type that falls under Tag\n",
+    "html = list(soup.children)[0]\n",
+    "\n",
+    "# get the body tag\n",
+    "body = list(html.children)[3]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# all products in card are within this div tag\n",
+    "products = body.findAll(\"div\",{\"class\":\"sc-list-item-content\"})\n",
+    "\n",
+    "# empty dictionary for all product info\n",
+    "appendix={}\n",
+    "\n",
+    "# loop through all products in the cart and get their link, item code, name, price\n",
+    "for each in products:\n",
+    "    link = \"https://www.amazon.ca\" + each.div.find(\"a\",{\"class\":\"a-link-normal sc-product-link\"})['href']\n",
+    "    \n",
+    "    sub_code= re.sub(r'^.*?product/', '', link)\n",
+    "    code = sub_code[:sub_code.find(\"/ref\")]\n",
+    "    \n",
+    "    item_link_response = requests.get('https://api.rainforestapi.com/request?api_key=B4057000DCF0420D8944FE2D7CFCF6B9&type=product&amazon_domain=amazon.ca&asin=' + code)\n",
+    "    name = re.sub('\\s+',' ',each.div.find(\"span\",{\"class\":\"a-size-medium sc-product-title a-text-bold\"}).text)\n",
+    "    price = re.sub('\\s+',' ',each.div.find(\"span\",{\"class\":\"a-size-medium a-color-base sc-price sc-white-space-nowrap sc-product-price a-text-bold\"}).text)\n",
+    "    \n",
+    "    category = []\n",
+    "    for each in item_link_response.json()['product']['categories']:\n",
+    "        category.append(each['name'])\n",
```

Comment:
in terms of figuring out what level of category to put, i think the first layer is enough
E.g. Electronics is great, no need for > Headphones > wireless or further breakdown
we can keep it simple with the basic categories as the graphs on our tool dashboard would get too complicated with more specific categories

so as you mentioned in your initial comment, the index 0 "umbrella" category is sufficient, we can turn that into a variable instead of an array